src/components/Liveness/index.js

```diff
@@ -34,31 +41,61 @@ export default class Liveness extends Component<CameraType, State> {
 
   loadChallenges = () => {
     this.setState({...initialState})
-    getLivenessChallenges(
-      challenges =>  this.setState({ challenges, hasLoaded: true }),
-      () => this.setState({ hasLoaded: true, hasError: true })
-    )
+    const options = {
+      endpoint: `${process.env.FACE_TORII_URL || ''}/challenge`,
+      contentType: 'application/json',
+      token: `Bearer ${this.props.token}`
+    }
+    performHttpReq(options, this.handleResponse, this.handleError)
+  }
+
+  handleResponse = (response: Object) => {
+    const {challenge, id} = response.data
+    this.setState({ challenges: challenge, id, hasLoaded: true })
+  }
+
+  handleError = () => {
+    this.setState({ hasLoaded: true, hasError: true })
+  }
+
+  handleChallengeSwitch = () => {
+    if (this.state.startedAt) {
+      this.setState({ switchSeconds: now() - this.state.startedAt })
+    }
+  }
+
+  handleVideoRecordingStart = () => {
+    this.setState({ startedAt: now() })
   }
 
   handleVideoRecorded = (blob: ?Blob) => {
-    this.props.onVideoRecorded(blob, this.state.challenges)
+    const { challenges, id, switchSeconds } = this.state
+    this.props.onVideoRecorded(blob, {
+      challenges, id, switchSeconds,
+    })
   }
 
   render() {
     const { i18n = {} } = this.props
     const { hasLoaded, hasError, challenges } = this.state
 
     return (
-      hasLoaded ?
-        hasError ?
-          <Error {...{error: serverError, i18n}} /> :
-          <LivenessCamera {...{
-            ...this.props,
-            challenges,
-            onRedo: this.loadChallenges,
-          }} />
-        :
-          <Spinner />
+      <div>{
+        hasLoaded ?
+          hasError ?
```

Comment:
won't push hard on it, but I would personally move this logic out
I find double ternaries hard to read